WordPress/Classes/ViewRelated/Site Creation/Design Selection/SiteDesignContentCollectionViewController.swift

```diff
@@ -176,22 +177,36 @@ class SiteDesignContentCollectionViewController: CollapsableHeaderViewController
     private func fetchSiteDesigns() {
         isLoading = true
 
-        SiteDesignSectionLoader.fetchSections(vertical: creator.vertical) { [weak self] result in
+        guard let sectionAssembler = sectionAssembler else {
             DispatchQueue.main.async {
-                guard let self = self else { return }
-
-                switch result {
-                case .success(let sections):
-                    self.dismissNoResultsController()
-                    self.sections = sections
-                    self.setupHelperView()
-                case .failure(let error):
-                    self.handleError(error)
+                SiteDesignSectionLoader.buildAssembler { [weak self] result in
+                    guard let self = self else { return }
+
+                    switch result {
+                    case .success(let assembler):
+                        let sections = assembler(self.creator.vertical)
```

Comment:
I don't think Swift will allow us to do named parameters like:

`assembler(vertical: self.creator.vertical)` which is a bummer, but if this ever grows we could take an approach where we pass a request type or maybe a tuple.